### DISCUSSION TITLE
Are the Express JS examples still valid?

### DISCUSSION THREAD

**Discussion by talonx** [clerk/javascript#3438] [Support request - DO NOT OPEN HERE] [answered]
Re: [https://clerk.com/blog/how-to-authenticate-api-requests-with-clerk-express](https://clerk.com/blog/how-to-authenticate-api-requests-with-clerk-express) - and the related source code - [https://github.com/argotdev/express-authentication/tree/main](https://github.com/argotdev/express-authentication/tree/main)

I've been trying to get these examples to work, but invariably run into a "Missing Clerk Secret Key" even though it's in my .env.local file with the key CLERK_API_KEY.

What am I missing?

**Comment by talonx** [comment]
So I figured out that the Clerk SDK version is 
```
"@clerk/clerk-sdk-node": "^5.0.8",
```
in my local (it's the latest AFAIK)

whereas the examples have 
```
"@clerk/clerk-sdk-node": "^4.10.3",
```
The latter works, so it's most likely a change in the env var names. Can anybody point me towards the correct vars?

**Reply by LekoArts** [reply]
https://clerk.com/docs/upgrade-guides/core-2/node

**Reply by talonx** [reply, marked as answer]
> https://clerk.com/docs/upgrade-guides/core-2/node

Thank you. That is helpful. However, the example now also needs `CLERK_PUBLISHABLE_KEY` in the env, although I am not sure why. This is supposed to be required in the frontend only.

When I add both `CLERK_PUBLISHABLE_KEY` and `CLERK_SECRET_KEY` the backend works.

**Reply by manuelpoelzl** [reply]
@talonx stumbled across the same thing during the setup of my express app. I also wondered about the need of the publishable key in the backend, maybe it is needed to verify that the backend belongs to a certain frontend app or something like that?

**Reply by talonx** [reply]
@manuelpoelzl I'm not sure but it worked for me. I wish the maintainers would respond with a clear answer here. The docs/examples clearly need an update.

**Reply by talonx** [reply]
Were the docs updated?
